dispatcher/MongoConnect.py

```diff
@@ -504,8 +504,17 @@ def set_stop_time(self, number, detectors, force):
                                 'max': {'$max': '$rate'}}}
                     ]):
                     rate[doc['_id']] = {'avg': doc['avg'], 'max': doc['max']}
+                channels = set()
+                if 'tpc' in detectors:
+                    # figure out which channels weren't running
+                    readers = list(self.latest_status[detectors]['readers'].keys())
+                    for doc in self.collections['node_status'].find({'host': {'$in': readers}, 'number': int(number)}):
```

Comment:
Any chance this is going to return a None, I don't think so right unless we have a run that fails to start (maybe)? Otherwise might be worth a check to not make sure to not try `for doc in None`

Reply:
This is a `find`, not a `find_one`, so worst case it returns an empty cursor, but it'll never return `None`. Also, this is behind the check for the successful update of the run endtime, so if a run didn't start then we can't reach this line.

Reply:
right, thanks